test_apps/smokeTests/gridTools.cpp

```diff
@@ -0,0 +1,436 @@
+    size_t &disagreements      // Counter for when AccessIJK() and GetValue() disagree
+) {
+    rms   = 0.f;
+    disagreements  = 0;
+    numMissingValues = 0;
+
+    std::vector< size_t > dims = grid->GetDimensions();
+    size_t x = dims[X];
+    size_t y = dims[Y];
+    size_t z = 1;
+    if ( dims.size() == 3 ) 
+        z = dims[Z];
+ 
+    double peak  = 0.f;
+    double sum     = 0;
+    for ( size_t k=0; k<z; k++ ) {
+        for ( size_t j=0; j<y; j++ ) {
+            for ( size_t i=0; i<x; i++ ) {
+                size_t indices[3] = { i, j, k };
+                //float trueValue = grid.AccessIJK( i, j, k );
+                double trueValue = grid->GetValueAtIndex( {i,j,k} );
+                
+                double coords[3];
+                grid->GetUserCoordinates( indices, coords );
+                float sampleValue = grid->GetValue( coords );
+                
+                if (sampleValue == grid->GetMissingValue()) {
+                    numMissingValues++;
+                    continue;
+                }
+
+                double error = abs( sampleValue - trueValue );
+
+                if (error != 0) {
+                    disagreements++;
+                }
+
+                if ( error > peak ) 
+                    peak = error;
+                sum += error*error;
+
+            }
+        }
+    }
+
+    rms = sqrt( sum / (x*y*z) );
+
+    if ( rms != 0 || disagreements > 0 )
+        return 1;
+    return 0;
+}
+
+size_t TestNodeIterator( const Grid *g, int& count, double& time ) {
+    Grid::ConstNodeIterator itr;
+    Grid::ConstNodeIterator enditr = g->ConstNodeEnd();
+
+    double t0 = Wasp::GetTime();
+    itr = g->ConstNodeBegin();
+
+    for ( ; itr!=enditr; ++itr) {
+        count++;
+    }
+
+    size_t expectedCount = 1;
+    std::vector< size_t > dims = g->GetDimensions();
+    for ( auto dim : dims )
+        expectedCount *= dim;
+
+    time = Wasp::GetTime() - t0;
+
+    return expectedCount;
+}
+
+void PrintStats( double rms, size_t numMissingValues, size_t disagreements ) {
+    cout << "    RMS error:                                     " << rms << endl;
+    cout << "    Missing value count:                           " << numMissingValues << endl;
+    cout << "    AccessIJK() vs GetValue() disagreement count:  " << disagreements << endl;
+    cout << endl;
+}
+
+int RunTest(
+    Grid* grid
+) {
+    int rc=0;
+    double rms;
+    size_t numMissingValues;
+    size_t disagreements;
+
+    if ( CompareIndexToCoords( grid, rms, numMissingValues, disagreements ) ) {
+        cout << "       *** Error reported in " << grid->GetType() << " grid***" << endl;
+        rc = 1;
+    }
+
+    if ( grid->GetInterpolationOrder() == 0 )
+        cout << "  nearestNeighbor: " << endl;
+    else
+        cout << "  linear:          " << endl;
+    PrintStats( rms, numMissingValues, disagreements );
+
+    return rc;
+}
+
+int RunTests( 
```

Comment:
Follow VAPOR conventions for return status; use booleans if errors aren't logged via MyBase

Reply:
Fixed.